build-logic/src/main/kotlin/Mpp.kt

```diff
@@ -113,7 +114,11 @@ fun Project.configureMppTestsDefaults(withJs: Boolean = true) {
         nodejs()
       }
     }
-    macosX64("apple")
+    if (hostArchitecture() == "arm64") {
+      macosArm64("apple")
+    } else {
+      macosX64("apple")
+    }
```

Comment:
Could we just register both? I'm always a bit concerned this might break stuff like build cache or configuration cache (if we ever wanted to enable it in the project)

Reply:
Good point! I knew there was probably a simpler way 😅

Reply:
Not sure how much simpler it'll be because now we're going to have to lie to IntelliJ to have both ARM and Intel use the same source set... 😅

Reply:
Ah yes, just like that 👍 LGTM!

Reply:
Oh well maybe not 🙃 . Do we still have IDE support with this? (Life is hard...)

Reply:
This works because it's the `commonTest` SourceSet. The problem with `macos("macosArm64")` is that it won't recognize `appleTest` as a source set like `macos("apple")` did. 

See HttpEngineTest.kt for an example

<img width="1508" alt="Screenshot 2022-03-24 at 00 37 45" src="https://user-images.githubusercontent.com/3974977/159871395-d9d9a7d5-e386-44b7-8a93-dbc433f43d13.png">

Until we have HMPP, I don't see a way around the `if (sync)` workaround